File: oop/labaratories/labaratory-work-2/Models/Dish.cs
namespace labaratory_work_2.Models
{
    public abstract class Dish
    {
        public string Name { get; set; }
        public int Weight { get; set; }
        public double Price { get; set; }

        public Dish()
        {
            Name = "dish";
            Weight = 100;
            Price = 10;
        }

        public Dish(string name)
        {
            Name = name;
            Weight = 200;
            Price = 20;
        }

        public Dish(string name, int weight, double price = 15)
        {
            Name = name;
            Weight = weight;
            Price = price;
        }

        public abstract string DisplayInfo();
        public abstract double CalcFullPrice();
    }
}

File: oop/labaratories/labaratory-work-1/dish.cpp
#include "dish.h"

std::string Dish::GetName() const { return _name; }
int			Dish::GetWeight() const { return _weight; }
double		Dish::GetPrice() const { return _price; }

void Dish::SetName(std::string name) { _name = name; }
void Dish::SetWeight(int weight) { _weight = weight; }
void Dish::SetPrice(double price) { _price = price; }
